UefiDbgExt/memory.cpp

```diff
@@ -319,63 +358,70 @@ advlog (
-    if (EndAddress < EntryAddress) {
-      dprintf ("Looped logs not yet implemented in extension!\n");
-      Result = ERROR_NOT_SUPPORTED;
-      goto Exit;
-    } else {
-      // non-loop optimization, only download through the last message.
-      LogBufferSize = min (LogBufferSize, (ULONG)(EndAddress - InfoAddress));
-    }
-
-    LogBuffer = (CHAR *)malloc (LogBufferSize);
+  } else if (Version == 5) {
+    EntryAddress = InfoAddress + Info.LogBufferOffset;
+    EndAddress   = InfoAddress + Info.LogCurrentOffset;
+  } else {
+    dprintf ("\nVersion not implemented in debug extension!\n");
+    Result = ERROR_NOT_SUPPORTED;
+    goto Exit;
+  }
 
-    // Output() forces IO flush to inform user.
-    g_ExtControl->Output (DEBUG_OUTPUT_NORMAL, "Reading log (0x%x bytes) ... \n", LogBufferSize);
-    ReadMemory (InfoAddress, LogBuffer, LogBufferSize, &BytesRead);
-    if (BytesRead != LogBufferSize) {
-      dprintf ("Failed to read log memory!\n");
-      Result = ERROR_BAD_LENGTH;
-      goto Exit;
-    }
+  if (EndAddress < EntryAddress) {
+    dprintf ("Looped logs not yet implemented in extension!\n");
+    Result = ERROR_NOT_SUPPORTED;
+    goto Exit;
+  } else {
+    // non-loop optimization, only download through the last message.
+    LogBufferSize = min (LogBufferSize, (ULONG)(EndAddress - InfoAddress));
+  }
 
-    Offset = (ULONG)(EntryAddress - InfoAddress);
-    End    = (ULONG)(EndAddress - InfoAddress);
+  LogBuffer = (CHAR *)malloc (LogBufferSize);
 
-    dprintf ("\n------------------------------------------------------------------------------\n");
-    BOOLEAN  PrevNL = TRUE;
-    while (Offset < End) {
-      Entry = (ADVANCED_LOGGER_MESSAGE_ENTRY_V2 *)(LogBuffer + Offset);
-      if (Entry->Signature != 0x324d4c41) {
-        dprintf ("\nBad message signature!!\n");
-        break;
-      }
+  // Output() forces IO flush to inform user.
+  g_ExtControl->Output (DEBUG_OUTPUT_NORMAL, "Reading log (0x%x bytes) ... \n", LogBufferSize);
+  ReadMemory (InfoAddress, LogBuffer, LogBufferSize, &BytesRead);
+  if (BytesRead != LogBufferSize) {
+    dprintf ("Failed to read log memory!\n");
+    Result = ERROR_BAD_LENGTH;
+    goto Exit;
+  }
 
-      ULONG  StringEnd = Offset + Entry->MessageOffset + Entry->MessageLen;
-      CHAR   Temp      = LogBuffer[StringEnd];
-      LogBuffer[StringEnd] = 0;
+  Offset = (ULONG)(EntryAddress - InfoAddress);
+  End    = (ULONG)(EndAddress - InfoAddress);
 
-      if (PrevNL) {
-        dprintf (
-          "%-8s| %-8s| ",
-          (Entry->Phase < PHASE_COUNT ? PhaseStrings[Entry->Phase] : "UNK"),
-          ErrorLevelToString (Entry->DebugLevel)
-          );
-      }
+  dprintf ("\n------------------------------------------------------------------------------\n");
+  BOOLEAN  PrevNL = TRUE;
 
-      dprintf ("%s", LogBuffer + Offset + Entry->MessageOffset);
-      PrevNL = (LogBuffer[StringEnd - 1] == '\n');
+  while (Offset < End) {

```

Comment:
```suggestion
  while (Offset <= End) {

```

Reply:
Probably also needs the offset of the structure. Will add this.